src/library_wasmfs_opfs.js

```diff
@@ -0,0 +1,250 @@
+    }
+    _emscripten_proxy_finish(ctx);
+  },
+
+  // Return the file ID for the file with `name` under `parent`, creating it if
+  // it doesn't exist and `create` or otherwise return one of the following
+  // error codes:
+  //
+  // -1: file does not exist.
+  // -2: file exists but it is actually a directory.
+  // -3: file exists but an access handle cannot be created for it.
+  $wasmfsOPFSGetOrCreateFile__deps: ['$wasmfsOPFSAllocate',
+                                     '$wasmfsOPFSDirectories',
+                                     '$wasmfsOPFSFiles'],
+  $wasmfsOPFSGetOrCreateFile: async function(parent, name, create) {
+    let parentHandle = wasmfsOPFSDirectories.get(parent);
+    assert(parentHandle !== undefined);
+    let fileHandle;
+    try {
+      fileHandle = await parentHandle.getFileHandle(name, {create: create});
+    } catch (e) {
+      if (e.name === "NotFoundError") {
+        return -1;
+      }
+      if (e.name === "TypeMismatchError") {
+        return -2;
+      }
+      throw e;
+    }
+    return wasmfsOPFSAllocate(wasmfsOPFSFiles, fileHandle);
+  },
+
+  // Return the file ID for the directory with `name` under `parent`, creating
+  // it if it doesn't exist and `create` or otherwise one of the following error
+  // codes:
+  //
+  // -1: directory does not exist.
+  // -2: directory exists but is actually a data file.
+  $wasmfsOPFSGetOrCreateDir__deps: ['$wasmfsOPFSAllocate',
+                                    '$wasmfsOPFSDirectories'],
+  $wasmfsOPFSGetOrCreateDir: async function(parent, name, create) {
+    let parentHandle = wasmfsOPFSDirectories.get(parent);
+    assert(parentHandle !== undefined);
+    let childHandle;
+    try {
+      childHandle =
+          await parentHandle.getDirectoryHandle(name, {create: create});
+    } catch (e) {
+      if (e.name === "NotFoundError") {
+        return -1;
+      }
+      if (e.name === "TypeMismatchError") {
+        return -2;
+      }
+      abort("Unknown exception " + e.name);
+    }
+    return wasmfsOPFSAllocate(wasmfsOPFSDirectories, childHandle);
+  },
+
+  _wasmfs_opfs_get_child__deps: ['$wasmfsOPFSGetOrCreateFile',
+                                 '$wasmfsOPFSGetOrCreateDir'],
+  _wasmfs_opfs_get_child:
+      async function(ctx, parent, namePtr, childTypePtr, childIDPtr) {
+    let name = UTF8ToString(namePtr);
+    let childType = 1;
+    let childID = await wasmfsOPFSGetOrCreateFile(parent, name, false);
+    if (childID == -2) {
+      childType = 2;
+      childID = await wasmfsOPFSGetOrCreateDir(parent, name, false);
+    }
+    {{{ makeSetValue('childTypePtr', 0, 'childType', 'i32') }}};
+    {{{ makeSetValue('childIDPtr', 0, 'childID', 'i32') }}};
+    _emscripten_proxy_finish(ctx);
+  },
+
+  _wasmfs_opfs_get_entries__deps: [],
+  _wasmfs_opfs_get_entries: async function(ctx, dirID, entries) {
+    let dirHandle = wasmfsOPFSDirectories.get(dirID);
+    for await (const [name, child] of dirHandle.entries()) {
```

Comment:
I think we stay away from `const` since it is more verbose than `var` (and `let`). Looks like a few snuck into `library_webgl` and `library_webgpu` but that's it, so probably best not to add more atm.